src/plugins/graph/components/simple-attribute-label.tsx

```diff
@@ -1,42 +1,53 @@
-import React, {useRef} from "react";
+import React, {useEffect, useRef, useState} from "react";
 import {createPortal} from "react-dom";
 import {observer} from "mobx-react-lite";
 import {GraphPlace } from "../imports/components/axis-graph-shared";
 import {AttributeType} from "../../../models/data/attribute";
 import {AxisOrLegendAttributeMenu} from "../imports/components/axis/components/axis-or-legend-attribute-menu";
-import { graphPlaceToAttrRole, kGraphClassSelector } from "../graph-types";
 import { useDataConfigurationContext } from "../hooks/use-data-configuration-context";
 import { useGraphModelContext } from "../models/graph-model";
 import { IDataSet } from "../../../models/data/data-set";
 import DropdownCaretIcon from "../dropdown-caret.svg";
+import { useGraphElementIdContext } from "./graph";
 
 import "../components/legend/multi-legend.scss";
 
 interface ISimpleAttributeLabelProps {
   place: GraphPlace
-  onChangeAttribute?: (place: GraphPlace, dataSet: IDataSet, attrId: string) => void
+  index: number
+  attrId: string
+  onChangeAttribute?: (place: GraphPlace, dataSet: IDataSet, attrId: string, oldAttrId?: string) => void
   onRemoveAttribute?: (place: GraphPlace, attrId: string) => void
   onTreatAttributeAs?: (place: GraphPlace, attrId: string, treatAs: AttributeType) => void
 }
 
 export const SimpleAttributeLabel = observer(
   function SimpleAttributeLabel(props: ISimpleAttributeLabelProps) {
-    const {place, onTreatAttributeAs, onRemoveAttribute, onChangeAttribute} = props;
+    const {place, index, attrId, onTreatAttributeAs, onRemoveAttribute, onChangeAttribute} = props;
     const simpleLabelRef = useRef<HTMLDivElement>(null);
-    const parentElt = simpleLabelRef.current?.closest(kGraphClassSelector) as HTMLDivElement ?? null;
+    const [readyForMenu, setReadyForMenu] = useState<boolean>(false);
     const dataConfiguration = useDataConfigurationContext();
+    const graphElementId = useGraphElementIdContext();
```

Comment:
The original code was:
```
const simpleLabelRef = useRef<HTMLDivElement>(null);
const parentElt = simpleLabelRef.current?.closest(kGraphClassSelector) as HTMLDivElement ?? null;
```
The Ref is null on first render causing the menu to not get immediately created. Since changing a Ref does not cause a re-render, there were cases where the menu _never_ got created.

In diagnosing the problem, I found [documentation](https://react.dev/learn/referencing-values-with-refs#best-practices-for-refs) stating that Refs' `current` should not be referenced directly in render methods, so I landed on the implementation here.  I'm open to better ideas, though - it does seem a bit clunky.

Reply:
Updated based on our conversation.  Essentially using old strategy, but with a State rather than a Ref so that the second render is guaranteed.  Simpler, and fixes the non-rendering menus issue that the old implementation had when menus are dynamic.